core/dbt/parser/base.py

```diff
@@ -95,6 +96,41 @@ def get_schema(_):
         self._get_schema_func = get_schema
         return self._get_schema_func
 
+    def get_alias_func(self):
+        """The get_alias function is set by a few different things:
+            - if there is a 'generate_alias_name' macro in the root project,
+                it will be used.
+            - if that does not exist but there is a 'generate_alias_name'
+                macro in the 'dbt' internal project, that will be used
+            - if neither of those exist (unit tests?), a function that returns
+                the 'default alias' as set in the model's filename or alias
+                configuration.
+        """
+        if self._get_alias_func is not None:
+            return self._get_alias_func
+
+        get_alias_macro = self.macro_manifest.find_macro_by_name(
+            'generate_alias_name',
+            self.root_project_config.project_name
+        )
+        if get_alias_macro is None:
+            get_alias_macro = self.macro_manifest.find_macro_by_name(
+                'generate_alias_name',
+                GLOBAL_PROJECT_NAME
+            )
+        if get_alias_macro is None:
+            def get_alias(node, _):
+                return node.name
```

Comment:
@drewbanin, I'm not sure what I should be returning here? I don't think it's `node.name`.

Reply:
The `get_alias_macro` should really never be `None` -- this is just an extra safety measure I think. Returning `node.name` is probably the move here -- we do something similar in the `generate_alias_name` macro [here](https://github.com/fishtown-analytics/dbt/pull/1363/files/298dde7eb3fe618b80f24598e50aac950db37d1f#diff-aa6cb00fd04708e215636ddeb04393faR18).

I could also buy the argument that we should return `node.alias` here, but we're in the code that's supposed to _determine_ the alias! So, in the event that something really bad happened and there's no `generate_alias_name` macro in the context at all, i think `name` is a reasonable fallback

Reply:
I think it should default to the old behavior, which was this:
```
config.config.get('alias', parsed_node.get('name'))
```

I think I would have the function duplicate the old behavior. Something like this, duplicating the logic in the default macro, should be equivalent:

```
def get_alias(node, custom_alias_name=None):
    if custom_alias_name is None:
        return node.name
    else:
        return custom_alias_name
```